node/testing/src/scenarios/multi_node/connection_discovery.rs

```diff
@@ -51,28 +51,26 @@ impl RustNodeAsSeed {
             ocaml::Node::spawn(18302, 18303, 18301, &dir.join("ocaml1"), [&rust_node_ma]).unwrap();
         let ocaml_peer_id1 = ocaml_node1.peer_id().into();
 
-        let mut peers = vec![ocaml_peer_id0, ocaml_peer_id1];
+        let mut unknown_peers = vec![&ocaml_peer_id0, &ocaml_peer_id1];
         let mut duration = Duration::from_secs(8 * 60);
-        while !peers.is_empty() {
+        while !unknown_peers.is_empty() {
             // wait for ocaml node to connect
-            let connected = driver
+            let event = driver
                 .wait_for(
                     duration,
-                    connection_finalized_event(|peer| peers.contains(peer)),
+                    route_added_event(|peer| unknown_peers.contains(&peer)),
                 )
                 .await
                 .unwrap()
                 .expect("expected connected event");
-            let (ocaml_peer, _) = as_connection_finalized_event(&connected.1).unwrap();
-            peers.retain(|peer| peer != ocaml_peer);
-            let ocaml_peer = ocaml_peer.clone();
+            if let Some((ocaml_peer, _)) = as_route_added_event(&event.1) {
+                unknown_peers.retain(|peer| *peer != ocaml_peer);
+            } else {
+                assert!(false, "unexpecte action");
+            }
             // execute it
-            let state = driver.exec_step(connected).await.unwrap().unwrap();
+            let _ = driver.exec_step(event).await.unwrap().unwrap();
             // check that now there is an outgoing connection to the ocaml peer
```

Comment:
Double unwrap? Why it can't fail?

---
Dangling comment:
```suggestion
```
I'd remove all comments in this function because they could turn into lying ones.